homeassistant/components/lgthinq/switch.py

```diff
@@ -0,0 +1,197 @@
+"""Support for switch entities."""
+
+from __future__ import annotations
+
+from collections.abc import Collection
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from thinqconnect import PROPERTY_WRITABLE, DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration.homeassistant.property import create_properties
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .const import POWER_OFF, POWER_ON, THINQ_DEVICE_ADDED
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import PropertyInfo, ThinQEntity, ThinQEntityDescription
+
+
+# Functions for switch operation.
+def value_to_power_state_converter(value: Any) -> str:
+    """Convert the value to string that represents power state."""
+    return POWER_ON if bool(value) else POWER_OFF
+
+
+@dataclass(kw_only=True, frozen=True)
+class ThinQSwitchEntityDescription(ThinQEntityDescription, SwitchEntityDescription):
+    """The entity description for switch."""
+
+
+OPERATION_SWITCH_DESC: dict[ThinQProperty, ThinQSwitchEntityDescription] = {
+    ThinQProperty.AIR_FAN_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.AIR_FAN_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
```

Comment:
Name can be removed since you have translation keys.

If during testing you don't see the translated name, you probably forgot to generate translations. I don't have the command at me right now, let me know if you need it so you don't have to search yourself

Reply:
Modified